sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -133,8 +136,27 @@ private object CosmosConfig {
     userProvidedOptions: Map[String, String] // user provided config
   ) : Map[String, String] = {
 
+    var accountDataResolverCls: AccountDataResolver = null
+    val serviceLoader = ServiceLoader.load(classOf[AccountDataResolver])
```

Comment:
this is loading ServiceLoader per configuration.

this should be loaded once per JVM (statically) and be reused for all config resolutions.

Reply:
you are invoking `ServiceLoader.load(classOf[AccountDataResolver])` inside another method call, which means per method call this will be invoked again.

ServiceLoader.load(classOf[AccountDataResolver]) should be invoked only once and you should store for future access

Reply:
As discussed offline, we will be skipping this for now.